contracts/upgradeable/EIP712CrossChainUpgradeable.sol

```diff
@@ -1,27 +1,35 @@
 // SPDX-License-Identifier: GPL-3.0-only
 pragma solidity ^0.8.0;
 
-import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/ECDSAUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 
 /**
  * @dev https://eips.ethereum.org/EIPS/eip-712[EIP 712] is a standard for hashing and signing of typed structured data.
  *
  * This contract is based on OpenZeppelin's implementation:
- * https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.8.0/contracts/utils/cryptography/EIP712.sol
+ * https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/master/contracts/utils/cryptography/EIP712Upgradeable.sol
  *
  * NOTE: Modified version that allows to build a domain separator that relies on a different chain id than the chain this
  * contract is deployed to. An example use case we want to support is:
  * - User A signs a message on chain with id = 1
  * - User B executes a method by verifying user A's EIP-712 compliant signature on a chain with id != 1
  */
-abstract contract EIP712CrossChain {
+abstract contract EIP712CrossChainUpgradeable is Initializable {
     /* solhint-disable var-name-mixedcase */
-    bytes32 private immutable _HASHED_NAME;
-    bytes32 private immutable _HASHED_VERSION;
-    bytes32 private immutable _TYPE_HASH;
+    bytes32 private _HASHED_NAME;
+    bytes32 private _HASHED_VERSION;
+    bytes32 private constant _TYPE_HASH = keccak256("EIP712Domain(string name,string version,uint256 chainId)");
```

Comment:
Should this be assigned in initialize()?

https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#avoid-initial-values-in-field-declarations